scipy/special/tests/test_basic.py

```diff
@@ -1824,10 +1824,12 @@ def test_nan_inputs(self, x, exact):
 
     def test_mixed_nan_inputs(self):
         x = np.array([np.nan, 1, 2, 3, np.nan])
-        result = special.factorial(x, exact=True)
-        assert_equal(np.array([np.nan, 1, 2, 6, np.nan]), result)
-        result = special.factorial(x, exact=False)
-        assert_equal(np.array([np.nan, 1, 2, 6, np.nan]), result)
+        with suppress_warnings() as sup:
+            sup.filter(DeprecationWarning, "Using factorial() with floats is deprecated")
```

Comment:
Does this need `\\(\\)`?

Reply:
Looks like it. Didn't test locally; this warning only showed up in 3.9-dev so I assume something is changing in deprecation warnings default behavior.